tuning/utils/merge_model_utils.py

```diff
@@ -0,0 +1,91 @@
+# Standard
+from typing import Union
+import argparse
+import json
+import os
+
+# Third Party
+from peft import PeftModel
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+def create_merged_model(
+    checkpoint_models: Union[str, list[str]],
+    export_path: str = None,
+    base_model: str = None,
+    save_tokenizer: bool = True,
+):
+    """Given a base model & checkpoint model(s) which were tuned with lora, load into memory
+    & create a merged model. If an export path is specified, write it to disk. If multiple
+    checkpoint models are provided, we merge_and_unload() them one after the other, which
+    combines them with equal weights.
+
+    TODO: In the future, it's probably a good idea to explore different combination schemes,
+    which can likely be done using a combination of add_weighted_adapter() and merge_and_unload().
+
+    References:
+    - https://github.com/huggingface/peft/issues/1040
+    - https://github.com/huggingface/peft/issues/280#issuecomment-1500805831
+    - https://huggingface.co/docs/peft/main/en/package_reference/lora#peft.LoraModel.add_weighted_adapter
+
+    Args:
+        checkpoint_model: Union[str, list[str]]
+            One or more lora checkpoints containing adapters.
+        export_path: str
+            Path to export the merged model to.
+        base_model: str
+            Base model to be leveraged. If no base model is specified, the base model is pulled
+            from the checkpoint model's adapter config.
+        save_tokenizer: bool
+            Indicates whether or not we should save the tokenizer from the base model. Only
+            used if the export_path is set.
+
+    Returns:
+        transformers model
+            Merged model created from the checkpoint / base model.
+    """
+    if isinstance(checkpoint_models, str):
+        checkpoint_models = [checkpoint_models]
+
+    if base_model is None:
+        base_model = fetch_base_model_from_checkpoint(checkpoint_models)
```

Comment:
I was just running this, and `fetch_base_model_from_checkpoint()` expects a string but `checkpoint_models` is being set to a list here which is causing an error. Sorry for missing this before